### PR TITLE
Tweak list of optimization rules

### PR DESCRIPTION
**Pull request by Dandandan**
# Which issue does this PR close?

Closes #.

 # Rationale for this change

I ran all the optimization rules twice and looked at the change in output in the TPC-H regression tests.
Next, I removed all the rules that didn't have any effect, and tried to remove the earlier passes.

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/optimizer/src/optimizer.rs [hunk 1]
```diff
@@ -163,6 +162,14 @@ impl Optimizer {
         rules.push(Arc::new(FilterPushDown::new()));
         rules.push(Arc::new(LimitPushDown::new()));
         rules.push(Arc::new(SingleDistinctToGroupBy::new()));
+        rules.push(Arc::new(ProjectionPushDown::new()));
```
Comment (Dandandan) [inline datafusion/optimizer/src/optimizer.rs:165]: This one can simply be run after the other optimizations.

File: benchmarks/expected-plans/q21.txt [hunk 2]
```diff
@@ -7,12 +7,12 @@ Sort: numwait DESC NULLS FIRST, supplier.s_name ASC NULLS LAST
             Inner Join: l1.l_orderkey = orders.o_orderkey
               Inner Join: supplier.s_suppkey = l1.l_suppkey
                 TableScan: supplier projection=[s_suppkey, s_name, s_nationkey]
-                Filter: l1.l_receiptdate > l1.l_commitdate AND l1.l_receiptdate > l1.l_commitdate
+                Filter: l1.l_receiptdate > l1.l_commitdate
```
Comment (alamb) [inline benchmarks/expected-plans/q21.txt:10]: 👍

**Review by alamb — APPROVED** [review]
The new plans look a lot nicer to me. The only thing I start to worry about is how long our planning will take -- Perhaps it would be good to add some logging to improve the results.

Maybe related to https://github.com/apache/arrow-datafusion/issues/1160

**Comment by Dandandan** [comment]
> The new plans look a lot nicer to me. The only thing I start to worry about is how long our planning will take -- Perhaps it would be good to add some logging to improve the results.
> 
> Maybe related to #1160



> The new plans look a lot nicer to me. The only thing I start to worry about is how long our planning will take -- Perhaps it would be good to add some logging to improve the results.
> 
> Maybe related to #1160

Yes, good idea. Also some optimization rules might be improved to e.g. not create filters that already exist etc. But in the end we would also need to keep the passes as simple as possible - a hard balancing act :)

**Comment by andygrove** [comment]
I've just started reviewing this PR and noticed one difference in one of the queries I tested with that might not be optimal.
 
## before

Table scan filter: `CAST(col1 AS Int64) BETWEEN Int64(4) AND Int64(4) + Int64(3)`

## after

Table scan filter: `CAST(col1 AS Int64) >= Int64(4), CAST(col1 AS Int64) <= Int64(7)`

It looks like we are now performing the CAST twice.

**Comment by andygrove** [comment]
Actually, I am testing against the latest DF that includes this PR but also has other changes ... so this change could possibly be unrelated to this PR. I will post an update here soon.

**Pull request merged**